### PR TITLE
Fix get separators (#9)

### PR DESCRIPTION
**Pull request by angelaszek**

### CODE REVIEW

**Comment by angelaszek** [comment]
I made the two tweaks to improve the checking of the first line for the separators and making sure the version specified in the constructor is consistent with the version in the file. These changes are unit tested too. I am not sure if you wanted any changes or if we are good to merge this.

File: src/main/java/com/imsweb/x12/reader/X12Reader.java [hunk 1]
```diff
@@ -277,6 +287,43 @@ else if (loop.getLoop() != null)
...
+        String version = null;
+        if (c == segmentSeparator) {
+            String lineString = line.toString();
+            int versionStartPos = lineString.lastIndexOf(elementSeparator);
+            if (versionStartPos != -1)
+                version = lineString.substring(versionStartPos + 1, lineString.length());
+        }
+        reader.reset();
+
+        boolean result = (X091_ANSI_VERSION.equals(version) && FileType.ANSI835_4010_X091.equals(type)) ||
```
[the remark this answers is not in the document]
Comment (ctmay4) [reply src/main/java/com/imsweb/x12/reader/X12Reader.java]: My bad, I did't realize you were checking both version and type.  I don't think you have to make any change, but would a Map be a better solution?

```java
Map<FileType, String> types = new HashMap<>;
types.put(FileType.ANSI835_4010_X091, X091_ANSI_VERSION);
.
.
.
```

then you can just make sure for the type that the value in the map matches the type in the claim?

**Comment by angelaszek** [comment]
I made the changes so that it uses a map.

File: src/main/java/com/imsweb/x12/reader/X12Reader.java [hunk 2]
```diff
@@ -41,6 +41,17 @@
     private static int _COMPOSITE_SEPARATOR_POS = 104; // array position
     private static int _SEGMENT_SEPARATOR_POS = 105; // array position
 
+    private static final String X091_ANSI_VERSION = "004010X091A1";
```
Comment (ctmay4) [inline src/main/java/com/imsweb/x12/reader/X12Reader.java]: Private static variables should be prefixed by underscore.

File: src/main/java/com/imsweb/x12/reader/X12Reader.java [hunk 3]
```diff
@@ -41,6 +41,17 @@
...
+    private static final String X221_ANSI_VERSION = "005010X221A1";
+    private static final String X096_ANSI_VERSION = "004010X096A1";
+    private static final String X097_ANSI_VERSION = "004010X097A1";
+    private static final String X098_ANSI_VERSION = "004010X098A1";
+    private static final String X222_ANSI_VERSION = "005010X222A1";
+    private static final String X223_ANSI_VERSION = "005010X223A2";
+
+    private static final Map<FileType, String> TYPES = new HashMap<>();
```
Comment (ctmay4) [inline src/main/java/com/imsweb/x12/reader/X12Reader.java]: Should be prefixed with underscore.

File: src/main/java/com/imsweb/x12/reader/X12Reader.java [hunk 4]
```diff
@@ -170,7 +191,9 @@ private void parse(FileType type, Reader reader) throws IOException {
 
         // set up delimiters
         Separators separators = getSeparators(reader);
-        if (separators != null) {
+        boolean consistentVersions = checkVersionsAreConsistent(type, separators, reader);
+
+        if (separators != null && consistentVersions) {
```
Comment (ctmay4) [inline src/main/java/com/imsweb/x12/reader/X12Reader.java]: Why assign `consistentVersions` to a local variable?  Just do the check in the if statement.

File: src/main/java/com/imsweb/x12/reader/X12Reader.java [hunk 5]
```diff
@@ -277,6 +300,43 @@ else if (loop.getLoop() != null)
...
+        }
+        reader.reset();
+
+        boolean result = (FileType.ANSI835_4010_X091.equals(type) && TYPES.get(FileType.ANSI835_4010_X091).equals(version)) ||
+                (FileType.ANSI835_5010_X221.equals(type) && TYPES.get(FileType.ANSI835_5010_X221).equals(version)) ||
+                (FileType.ANSI837_4010_X096.equals(type) && TYPES.get(FileType.ANSI837_4010_X096).equals(version)) ||
+                (FileType.ANSI837_4010_X097.equals(type) && TYPES.get(FileType.ANSI837_4010_X097).equals(version)) ||
+                (FileType.ANSI837_4010_X098.equals(type) && TYPES.get(FileType.ANSI837_4010_X098).equals(version)) ||
+                (FileType.ANSI837_5010_X222.equals(type) && TYPES.get(FileType.ANSI837_5010_X222).equals(version)) ||
+                (FileType.ANSI837_5010_X223.equals(type) && TYPES.get(FileType.ANSI837_5010_X223).equals(version));
```
Comment (ctmay4) [inline src/main/java/com/imsweb/x12/reader/X12Reader.java]: The point of the `Map` was that you don't need this large boolean expression.  You just need to verify that the type in the file is that same as the type in the map.  No need to put specifiec cases.

```java
boolean result = TYPES.get(type).equals(version);
```

That will allow you to just add a new entry to the `Map` to get additional type support.

**Comment by ctmay4** [comment]
@angelaszek more code review issues.

**Comment by angelaszek** [comment]
I made the changes.

**Pull request merged**